### ISSUE TITLE
Auto minimal reproducer [medium term]

### ISSUE THREAD

**Issue by ajassani** [AMD-AGI/TraceLens#20]
When analyzing performance issues, we often identify specific PyTorch native operations (aten:: ops) that are slow on a given hardware/software environment. The next step is to create a minimal reproducible example (min-repro) to isolate and investigate the issue.

To get started, we have a sample script based on Yu-Yun Chang’s work that generates a min-repro for aten::add. It constructs tensors with custom shapes, strides, and dtypes, executes the operation, and benchmarks it. However, this script is specific to aten::add, and we need to generalize it to support arbitrary PyTorch native ops identified in profiling reports.

Scope
1. Current Focus: PyTorch native ops (aten:: ops).
1. Future Considerations: We can extend this approach to other operator backends later.
Expected Features
- Input: Operation name (aten::<op>), tensor shapes, strides, dtypes.
- Output: A Python script that:
Constructs tensors correctly (including as_strided views if needed).
Executes the specified operation.
Measures execution time with warmup and repeat runs.
- Automation: This process should be scriptable to streamline debugging workflows.

```python

import torch
import time

def generate_min_repro_py(filename, size_a, stride_a, dtype_a,
                          size_b, stride_b, dtype_b,
                          warmup=10, repeats=50):
    """
    Generates a minimal reproducible example Python script that
    creates two tensors with as_strided views and performs an elementwise-add.
    
    Parameters:
      filename (str): The name of the output .py file.
      size_a (tuple): The shape of tensor a.
      stride_a (tuple): The stride for tensor a.
      dtype_a (torch.dtype): The data type for tensor a.
      size_b (tuple): The shape of tensor b.
      stride_b (tuple): The stride for tensor b.
      dtype_b (torch.dtype): The data type for tensor b.
      warmup (int): Number of warmup iterations.
      repeats (int): Number of timed iterations.
    """
    # Convert torch dtypes to strings that can be used in code (e.g. "bfloat16")
    dtype_a_str = str(dtype_a).split('.')[-1]
    dtype_b_str = str(dtype_b).split('.')[-1]

    # Build the code string
    code = f'''import torch
import time

def run():
    # Create random tensors on CUDA with the specified dtypes.
    a = torch.randn({size_a}, device='cuda', dtype=torch.{dtype_a_str})
    b = torch.randn({size_b}, device='cuda', dtype=torch.{dtype_b_str})
    
    # Use as_strided to create the views with custom strides.
    a = torch.as_strided(a, size={size_a}, stride={stride_a})
    b = torch.as_strided(b, size={size_b}, stride={stride_b})
    
    # Warmup iterations
    for _ in range({warmup}):
        c = a + b
    torch.cuda.synchronize()
    
    # Timed iterations
    start = time.time()
    for _ in range({repeats}):
        c = a + b
    torch.cuda.synchronize()
    end = time.time()
    
    avg_time_ms = (end - start) / {repeats} * 1000
    print("Average kernel time: {{:.3f}} ms".format(avg_time_ms))
    
if __name__ == "__main__":
    run()
'''
    # Write the generated code to the file.
    with open(filename, "w") as f:
        f.write(code)
    print(f"Generated minimal repro file: {filename}")

# Example usage:
if __name__ == "__main__":
    # Here we pick one of the configurations.
    config = {
        'size_a': (5, 2048, 1536),
        'size_b': (5, 2048, 1536),
        'dtype_a': torch.bfloat16,
        'dtype_b': torch.bfloat16,
        'stride_a': (3145728, 1536, 1),
        'stride_b': (3145728, 1536, 1)
    }
    
    generate_min_repro_py(
        filename="min_repro.py",
        size_a=config['size_a'],
        stride_a=config['stride_a'],
        dtype_a=config['dtype_a'],
        size_b=config['size_b'],
        stride_b=config['stride_b'],
        dtype_b=config['dtype_b'],
        warmup=10,
        repeats=50
    )


```
